### DISCUSSION TITLE
Trainer special function "Sticks" keeping actual stick values intact

### DISCUSSION THREAD

**Discussion by raphaelcoeffic** [EdgeTX/edgetx#3162] [Ideas]
Currently (`2.8`), the trainer custom function "Sticks" does replace the stick values at a very early stage (while applying ADC calibration values), thus making it impossible to access these values while the trainer signal is valid.

I'm wondering whether it would not make more sense to replace the stick values used while evaluating inputs, thus leaving the stick values untouched.

Obviously, this would have some implications, for example when evaluating the throttle for timers (currently the stick value is used rather than the input value) or throttle warnings. These issues would need to be sorted out (not clear yet how).

**Comment by froqstar** [comment]
> when evaluating the throttle for timers (currently the stick value is used rather than the input value)

Isn't using the stick value rather than the input for throttle timers not great anyway? If you have a curve applied to the input that keeps the output quite low until ~75% throttle is applied (hockeystick curve) and a throttle percentage based timer, I would expect the percentage to be calculated _after_ applying the input curve, not before?
For throttle warning I am not so sure.

**Reply by raphaelcoeffic** [reply]
Yeah, I was thinking the same while checking the current code. That being said, there are still some challenges on how this can be changed in an acceptable way (keep storage compatibility).

**Comment by ChrisOhara57** [comment]
Please have a look at the proposal in issue #5532
